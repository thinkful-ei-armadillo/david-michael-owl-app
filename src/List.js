import React from 'react'

import Store from './STORE'

function List(){
    const people = Store.participants;
    // const peopleStatus = people.map(data =>
    //     { name: {people.name}}
    //     )

    return (
        <div className="People-list">
    
        {people.map((person) => 
            <div key={person.id}> 
              <img src = {person.avatar} alt="chat icon"/>
              <p>{person.name}</p>
              <p>{(person.onStage) ? 'On stage' : 'Off stage'}</p>
          </div>
        )}
        </div>
      )
    }


export default List;