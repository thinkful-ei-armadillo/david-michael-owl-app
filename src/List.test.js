import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import List from './List'

describe('<List />', () => {
    it('renders without crashing', ()=> {
        const div = document.createElement('div');
        ReactDOM.render(<List avatar ='Our header' name = 'Bob' onStage = {true} />, div);
        ReactDOM.unmountComponentAtNode(div);
      })
    })



describe('<List />', () => {
  it('renders without crashing', () => {
      // Render the component, as JSON
      const tree = renderer.create(<List avatar ='Our header' name = 'Bob' onStage = {true} />).toJSON();
      // Check whether it matches the previous snapshot
      // Stored in __snapshots__/App.test.js.snap
      expect(tree).toMatchSnapshot(); 
  });
});
