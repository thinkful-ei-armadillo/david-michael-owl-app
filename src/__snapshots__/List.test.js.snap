// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders without crashing 1`] = `
<div
  className="People-list"
>
  <div>
    <img
      alt="chat icon"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <p>
      Koren Templeton
    </p>
    <p>
      On stage
    </p>
  </div>
  <div>
    <img
      alt="chat icon"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    <p>
      Caty Flucker
    </p>
    <p>
      Off stage
    </p>
  </div>
  <div>
    <img
      alt="chat icon"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
    <p>
      Axe Kubicka
    </p>
    <p>
      Off stage
    </p>
  </div>
  <div>
    <img
      alt="chat icon"
      src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
    />
    <p>
      Frank Runciman
    </p>
    <p>
      Off stage
    </p>
  </div>
  <div>
    <img
      alt="chat icon"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <p>
      Ashla Attwool
    </p>
    <p>
      On stage
    </p>
  </div>
</div>
`;
